lib/src/protocol/kawa_h1/mod.rs

```diff
@@ -0,0 +1,1933 @@
+                return SessionResult::Continue;
+            } else {
+                return SessionResult::Close;
+            }
+        }
+
+        if self.response_stream.is_main_phase() {
+            self.frontend_readiness.interest.insert(Ready::WRITABLE);
+        }
+        if self.response_stream.is_terminated() {
+            metrics.backend_stop();
+            self.backend_stop = Some(Instant::now());
+            self.backend_readiness.interest.remove(Ready::READABLE);
+        }
+        SessionResult::Continue
+    }
+}
+
+impl<Front: SocketHandler, L: ListenerHandler + L7ListenerHandler> Http<Front, L> {
+    fn log_context(&self) -> LogContext {
+        LogContext {
+            request_id: self.context.id,
+            cluster_id: self.cluster_id.as_deref(),
+            backend_id: self.backend_id.as_deref(),
+        }
+    }
+
+    /// (code, reason)
+    pub fn get_response_line(&self) -> (Option<u16>, Option<&str>) {
+        if let kawa::StatusLine::Response { code, reason, .. } =
+            &self.request_stream.detached.status_line
+        {
+            let buffer = self.response_stream.storage.buffer();
+            let reason = reason
+                .data_opt(buffer)
+                .map(|data| unsafe { from_utf8_unchecked(data) });
+            (Some(*code), reason)
+        } else {
+            (None, None)
+        }
+    }
+
+    /// (method, authority, path, uri)
+    pub fn get_request_line(&self) -> (Option<Method>, Option<&str>, Option<&str>, Option<&str>) {
+        if let kawa::StatusLine::Request {
+            method,
+            authority,
+            path,
+            uri,
+            ..
+        } = &self.request_stream.detached.status_line
+        {
+            let buffer = self.request_stream.storage.buffer();
+            let method = method.data_opt(buffer).map(Method::new);
+            let authority = authority
+                .data_opt(buffer)
+                .map(|data| unsafe { from_utf8_unchecked(data) });
+            let path = path
+                .data_opt(buffer)
+                .map(|data| unsafe { from_utf8_unchecked(data) });
+            let uri = uri
+                .data_opt(buffer)
+                .map(|data| unsafe { from_utf8_unchecked(data) });
+            (method, authority, path, uri)
+        } else {
+            (None, None, None, None)
+        }
+    }
+
+    pub fn get_session_address(&self) -> Option<SocketAddr> {
+        self.context
+            .session_address
+            .or_else(|| self.frontend_socket.socket_ref().peer_addr().ok())
+    }
+
+    pub fn get_backend_address(&self) -> Option<SocketAddr> {
+        self.backend
+            .as_ref()
+            .map(|backend| backend.borrow().address)
+            .or_else(|| {
+                self.backend_socket
+                    .as_ref()
+                    .and_then(|backend| backend.peer_addr().ok())
+            })
+    }
+
+    fn protocol_string(&self) -> &'static str {
```

Comment:
please write a documenting comment, like "get the TLS protocol definition (if any)"